allennlp/common/util.py

```diff
@@ -57,8 +57,9 @@ def pad_sequence_to_length(sequence: List,
             padded_sequence.insert(0, default_value())
     return padded_sequence
 
+A = TypeVar('A')
 
-def add_noise_to_dict_values(dictionary: Dict[Any, float], noise_param: float) -> Dict[Any, float]:
+def add_noise_to_dict_values(dictionary: Dict[A, float], noise_param: float) -> Dict[A, float]:
```

Comment:
that's one reason. the other is that if you say the parameter has to be a `Dict[Any, float]` and then you try to give it a `Dict[str, float]` it won't type check, since you said that the input `dictionary` needs to accept `Any` key.

(if you want to get type-nerdy, functions are contravariant in input type and covariant in output type)